src/platform/with_wgl/native_gl_context.rs

```diff
@@ -54,10 +54,13 @@ impl Drop for NativeGLContext {
     fn drop(&mut self) {
         unsafe {
             if !self.weak {
+                self.unbind().unwrap(); //wglDeleteContext requires unbind
                 wgl::DeleteContext(self.render_ctx as *const _);
                 let window = user32::WindowFromDC(self.device_ctx);
-                user32::ReleaseDC(window, self.device_ctx);
-                user32::DestroyWindow(window);
+                if !window.is_null() {
```

Comment:
what has suddenly making `window` nullable? If it happened before and is something you've fixed, I guess that's fine.

But if it's because we can create multiple contexts associated with the same device context, and then have the window deleted, that can mean multiple things if I'm correct:

 * The context doesn't require the window to stay alive, in which case we can delete it earlier.
 * The context does require the window to stay alive, in which case this is wrong.
 * The context does require the window to stay alive at least until we do sharing? That doesn't make sense to me but could be it.

Can you tell me which one is the relevant one?